analysisservice/src/test/java/nu/yona/server/analysis/service/ActivityUpdateServiceTest.java

```diff
@@ -91,6 +94,8 @@
 	@Mock
 	private ActivityCacheService mockAnalysisEngineCacheService;
 	@Mock
+	private GoalRepository mockGoalRepository;
```

Comment:
The `ActivityUpdateService` doesn't have `GoalRepository` dependency. So why is this necessary?

Reply:
The method ``UserAnonymized.preloadGoals`` uses it and that's being called from the ``ActivityUpdateService``.